### PR TITLE
PICARD-1379: Fix PyUnicode_GetSize deprecation warning in astrcmp

### PR DESCRIPTION
**Pull request by phw**
# Summary

Use `PyUnicode_GetLength` instead of deprecated `PyUnicode_GetSize`. Requires Python >= 3.3, but we list Python 3.5 as the minimum anyway.

* This is a…
    * [ ] Bug fix
    * [ ] Feature addition
    * [x] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): [PICARD-1379](https://tickets.metabrainz.org/browse/PICARD-1379)

# Solution

# Action

### CODE REVIEW

File: picard/util/_astrcmp.c [hunk 1]
```diff
@@ -156,8 +156,8 @@ astrcmp(PyObject *self, PyObject *args)
 
 	us1 = PyUnicode_AS_UNICODE(s1);
 	us2 = PyUnicode_AS_UNICODE(s2);
```
Comment (zas) [inline picard/util/_astrcmp.c]: Isn't this also deprecated ?
https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_AS_UNICODE

Comment (phw) [reply picard/util/_astrcmp.c]: Indeed it is. There was just no warning output, probably because these are macros. I will update the PR

**Comment by phw** [comment]
I updated the module for the new Unicode API. For a quick speed comparison I measured running the unit test function 1 million times. On my system:

- Old implementation: 3.9155 seconds
- New implementation: 4.0373 seconds
- New implementation without https://github.com/metabrainz/picard/pull/1008/commits/6eb132ba3d133f3e2cc16c21eb37210d08b91483: 4.5491

So a slight performance decrease of 3%. But for comparison the Python implementation takes roughly 186 seconds for the same call count.

**Comment by phw** [comment]
Sorry, this needs more work. I just realized my patch breaks the threading

**Pull request closed without merging**